dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -62,13 +65,68 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         }
     }
 
+    // RegisterUnpackagedapplication
+    void PushNotificationManager::RegisterApplicationHelper(wil::unique_cotaskmem_string &unpackagedAppUserModelId, winrt::guid remoteId)
+    {
+            THROW_IF_FAILED(::CoInitializeEx(nullptr, COINITBASE_MULTITHREADED));
+
+            auto scopeExit = wil::scope_exit(
+                [&]() { CoUninitialize(); });
+
+            {
+                wil::com_ptr<INotificationsLongRunningPlatform> reunionEndpoint{
+                    wil::CoCreateInstance<NotificationsLongRunningPlatform, INotificationsLongRunningPlatform>(CLSCTX_LOCAL_SERVER) };
+
+                char processName[1024];
+                GetModuleFileNameExA(GetCurrentProcess(), NULL, processName, sizeof(processName) / sizeof(processName[0]));
+
+                winrt::throw_hresult(reunionEndpoint->RegisterFullTrustApplication(processName, remoteId, &unpackagedAppUserModelId));
+            }
+
+        return;
+    }
+
+    winrt::hresult PushNotificationManager::CreateChannelWithRemoteIdHelper(
+        wil::unique_cotaskmem_string& channelUri,
+        winrt::Windows::Foundation::DateTime& channelExpiryTime,
+        winrt::guid remoteId)
+    {
+        wchar_t appUserModelId[APPLICATION_USER_MODEL_ID_MAX_LENGTH + 1] = {};
```

Comment:
This doesn't look right. `APPLICATION_USER_MODEL_ID_MAX_LENGTH` already includes space for the null terminator.

Reply:
Yeah I missed it. Thanks.